fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -479,15 +491,16 @@ public String toString() {
     // Multi ref base table partition expression and ref slot ref map.
     @SerializedName(value = "partitionExprMaps")
     private Map<ExpressionSerializedObject, ExpressionSerializedObject> serializedPartitionExprMaps;
-    private Map<Expr, SlotRef> partitionExprMaps;
+    // Use LinedHashMap to keep the order of partition exprs by the user's defined order which can be used when mv contains multi
+    // partition columns.
+    private LinkedHashMap<Expr, SlotRef> partitionExprMaps;
+
     // ref base table to partition expression
-    private transient volatile Optional<Map<Table, Expr>> refBaseTablePartitionExprsOpt = Optional.empty();
+    private Optional<Map<Table, List<Expr>>> refBaseTablePartitionExprsOpt = Optional.empty();
     // ref bae table to partition column slot ref
-    private transient volatile Optional<Map<Table, SlotRef>> refBaseTablePartitionSlotsOpt = Optional.empty();
+    private Optional<Map<Table, List<SlotRef>>> refBaseTablePartitionSlotsOpt = Optional.empty();
```

Comment:
it seems that Expr used by partition-by-clause has exactly one SlotRef and the Expr is simple, so get SlotRef is not time-cosuming, why not just use Expr in these data structure, when corresponding SlotRef is required, just obtain it from the Expr directly on-the-fly.

Reply:
do we have a chance that a single partition-by expr is built from multiple SlotRef? for an example:
base table partition by (Country, Province, City, District), but MV's partitions  can be mapped in such a way partition by ws_concat('#', Country, Province, City, District).

Reply:
> it seems that Expr used by partition-by-clause has exactly one SlotRef and the Expr is simple, so get SlotRef is not time-cosuming, why not just use Expr in these data structure, when corresponding SlotRef is required, just obtain it from the Expr directly on-the-fly.

There are two reasons:
1. Even `SlotRef` compute is not time-consuming, we can only compute it once only to avoid repeating ...
2. `refBaseTablePartitionColumnsOpt` refers `refBaseTablePartitionExprsOpt` and it's called multi times, so needs to make it ready first.

Reply:
> do we have a chance that a single partition-by expr is built from multiple SlotRef? for an example:
base table partition by (Country, Province, City, District), but MV's partitions can be mapped in such a way partition by ws_concat('#', Country, Province, City, District).

Yes, we can support it, but it's still a huge work. But we may treat `ws_concat('#', Country, Province, City, District).` as a generated column to simplify the partition remapping. 

We can discuss this later if needed.